Scripts/Source/UnforgivingDevicesMain.psc

```diff
@@ -524,6 +537,9 @@ bool Function ActorIsValidForUD(Actor akActor)
     if loc_race.IsChildRace()    ;check that actor is not child
         return false
     endif
+    if (loc_actorbase.GetSex() == 0 && !AllowMenBondage)
```

Comment:
This part should also check if For Him is installed. Only if both For Him is installed and it is enabled, it should proceed. Also, please move the loc_actorbase.GetSex() == 0 on the end of if, so it is last condition. This way, it will not check actor sex, if the For Him is not installed (aka, optimization).

Reply:
Actually, I've put 
else
        ForHimInstalled = false
        AllowMenBondage = false
    endif

in the esp detection function, so it ensures it will be forced to be disabled, making this quite robust. So one less condition to check (as it's sure to be false if esp is not there), so maybe not good idea?

Swapping places - no problem, will do.

Reply:
That it true, but I think it should still be an other way around, as it makes code more clean to read. The ActorIsValidForUD is quite important function in UD, so it can be assumed that other people might trey to take a look at it in the source code. And if they see the condition as it is now, they will not realize that it is actually linked to For Him .esp. Also, checking bool is one of the fastest operations you can do in Papyrus, so it will not have that much of the impact.

So, I think the condition should look like this
if (!ForHimInstalled && !AllowMenBondage && loc_actorbase.GetSex() == 0)
    return false
endif

[This ](https://github.com/IHateMyKite/UnforgivingDevices/blob/8858c7014db0fcc0ba1283c9e25404503737ff75/Scripts/Source/UnforgivingDevicesMain.psc#L417) part will then be irrelevant

Reply:
Yep, this will look like it.
Guess we'll place 3 conditions check instead of two, if you insist, no problem.